library/choose_disk.py

```diff
@@ -353,46 +353,69 @@ def expand_disks(lookup_disks, ceph_type="", module=None):
     return final_disks
 
 
-def disk_label(partition):
+def disk_label(partition, current_fsid, ceph_disk):
     '''
     Reports if a partition is containing some ceph structures
     '''
-    # 1) let's search for legacy metadata made by ceph-disk
-    stdout = subprocess.check_output(["blkid", "-s", "PARTLABEL", "-o", "value", "{}".format(partition)])
 
+    # 1) let's search for metadata coming from ceph-volume
+    try:
+        output_cmd = subprocess.Popen(["ceph-volume", "lvm", "list", "--format=json", partition],
+                                      stdout=subprocess.PIPE)
+        raw_json, _ = output_cmd.communicate()
+        try:
+            json_dict = json.loads(raw_json)
+        except (ValueError, TypeError) as e:
+            fatal("Cannot parse ceph-volume properly : {}".format(e))
+
+        # If the disk have some metadata, the json is filled with information
+        for key, value in json_dict.items():
+            # As we parse a partition it should only have a single entry unless that's an error
+            if (len(value) > 1):
+                fatal("Error while parsing ceph-volume")
+            if "tags" in value[0]:
+                if "ceph.cluster_fsid" in value[0]['tags']:
+                    if value[0]["tags"]["ceph.cluster_fsid"] != current_fsid:
+                        return "foreign : {}".format(value[0]["tags"]["ceph.cluster_fsid"])
+            if "type" in value[0]:
+                return value[0]["type"]
+            else:
+                # ceph-volume should give a type
+                # If not, we don't really what it is _but_ the disks is used
+                # So we should ignore it, let's report it undefined
+                return "undefined"
+
+    except subprocess.CalledProcessError:
+        pass
+
+    # 2) let's try with ceph-disk
+    # If the disk have some metadata, the json is filled with information
+    for item in ceph_disk:
+        if "partitions" in item:
+            for partition_iter in item['partitions']:
+                if "path" in partition_iter:
+                    if partition_iter["path"] == partition:
+                        if "ceph_fsid" in partition_iter:
+                            if partition_iter["ceph_fsid"] != current_fsid:
+                                return "foreign : {}".format(partition_iter["ceph_fsid"])
+
+    # 3) let's search directly from the partitions labels
+    try:
+        stdout = subprocess.check_output(["blkid", "-s", "PARTLABEL", "-o", "value", "{}".format(partition)])
+    except subprocess.CalledProcessError:
```

Comment:
this is another instance of eating completely the stderr,stdout from the call, which is going to cause issues when trying to debug failures. `"failed"` is not sufficient here.

Reply:
Failed is not a debugging message but a label to say "this is not free but we don't understand what was wrong with it".
An extra debugging info could be added.